'eslint.runtime' binary executed with wrong cwd

Use case: to enable ESLint configs written in TypeScript, we should pass '--loader TS_LOADER_PACKAGE_OR_PATH' to Node runtime. Also, it would be nice to use host Node binary (not bundled with VSCode). 

Theoretically, we can use "eslint.runtime": "node" & "eslint.execArgv": ["--experimental-loader", "TS_LOADER_PACKAGE_OR_PATH"] settings combination. But problem is that 'runtime' binary executed with cwd: process.cwd() in client/src/client.ts:
``` typescript
const result: ServerOptions = {
	run: { module: serverModule, transport: TransportKind.ipc, runtime, options: { execArgv, cwd: process.cwd(), env } },
	debug: { module: serverModule, transport: TransportKind.ipc, runtime, options: { execArgv: execArgv !== undefined ? execArgv.concat(debugArgv) : debugArgv, cwd: process.cwd(), env } }
};
```

In my case, process.cwd() is '/', so Node just can't find TS loader installed in my project directory. Sure, absolute path can be used... but it machine dependent. And there is no ability to use something like "${WORKSPACE_DIR}/node-modules/ts-loader-path" in VSCode settings.

As a workaround, I use special script passed to extension via "eslint.runtime": ".vscode/spawn-node.js", then call 'process.chdir(dirname(fileURLToPath(import.meta.url)))' before spawn real Node process (which is required a bit magic with IPC, but it's out of scope). 

So, my proposal is to provide ability to use project directory as cwd on runtime execution. As additional parameter like 'eslint.executeRuntimeInWorkspaceDir', or by default - it doesn't matter.

Back at it trying to troubleshoot Windows.

Seems like for Windows, passing the script file as the `spawn` command will create `UKNOWN` error. Reference: https://stackoverflow.com/questions/43419893/how-do-i-fix-error-spawn-unknown-with-node-js-v7-8-0-on-windows-10

I'm not sure if it was the intention of the language server API, to allow a script command. So I tried the following:

```json
{
  "eslint.runtime": "node",
  "eslint.execArgv": ["./bin/eslint-runtime.js"],
}
```

This time I got around the previous error however, the entry point is resolved to working directory of VSCode. 
```sh
Error: Cannot find module 'C:\Users\psychobolt\AppData\Local\Programs\Microsoft VS Code\bin\eslint-runtime.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

So for Windows you'll have to end up pointing to the same issues as OP, having to use absolute path in the settings. The following works, if you need to pass additional NODE_OPTIONS.

```json
{
  "eslint.runtime": "node",
  "eslint.execArgv": ["D://User/my-project/bin/eslint-runtime.js"],
}
```

--- a/client/src/client.ts
+++ b/client/src/client.ts
@@ -380,6 +380,21 @@ export namespace ESLintClient {
 
 		return [client, acknowledgePerformanceStatus];
 
+		function getCurrentServerWorkingDirectory(): string {
+			if (!Workspace.isTrusted) {
+				return process.cwd();
+			}
+			const workspaceFolders = Workspace.workspaceFolders;
+			if (workspaceFolders === undefined || workspaceFolders.length !== 1) {
+				return process.cwd();
+			}
+			const uri = workspaceFolders[0].uri;
+			if (uri.scheme !== 'file') {
+				return process.cwd();
+			}
+			return uri.fsPath;
+		}
+
 		function createServerOptions(extensionUri: Uri): ServerOptions {
 			const serverModule = Uri.joinPath(extensionUri, 'server', 'out', 'eslintServer.js').fsPath;
 			const eslintConfig = Workspace.getConfiguration('eslint');
@@ -397,10 +412,11 @@ export namespace ESLintClient {
 				env = env || {};
 				env.NODE_ENV = nodeEnv;
 			}
+			const cwd = getCurrentServerWorkingDirectory();
 			const debugArgv = ['--nolazy', '--inspect=6011'];
 			const result: ServerOptions = {
-				run: { module: serverModule, transport: TransportKind.ipc, runtime, options: { execArgv, cwd: process.cwd(), env } },
-				debug: { module: serverModule, transport: TransportKind.ipc, runtime, options: { execArgv: execArgv !== undefined ? execArgv.concat(debugArgv) : debugArgv, cwd: process.cwd(), env } }
+				run: { module: serverModule, transport: TransportKind.ipc, runtime, options: { execArgv, cwd, env } },
+				debug: { module: serverModule, transport: TransportKind.ipc, runtime, options: { execArgv: execArgv !== undefined ? execArgv.concat(debugArgv) : debugArgv, cwd, env } }
 			};
 			return result;
 		}

